src/esa_apex_toolbox/upscaling/udp_job_manager.py

```diff
@@ -0,0 +1,146 @@
+        - `description` : Description of the job
+
+        """
+        if self.dataframe is None:
+            self.dataframe = jobs_dataframe
+        else:
+            raise ValueError("Jobs already added to the job manager.")
+
+    def start_job_thread(self):
+        """
+        Start running the jobs in a separate thread, returns afterwards.
+        """
+
+        udp_parameter_names = [p["name"] for p in self.udp_parameters()]
+
+        geojson_params = [p["name"] for p in self.udp_parameters() if
+                          p.get("schema", {}).get("subtype", "") == "geojson"]
+
+
+        output_file = Path("jobs.csv")
+        if self.dataframe is not None:
+            df = self._normalize_df(self.dataframe)
+
+            def normalize_fixed_param_value(name, value):
+                if isinstance(value, list) or isinstance(value, tuple):
+                    return len(df) * [value]
+                else:
+                    return value
+
+            new_columns = {
+                col: normalize_fixed_param_value(col,val) for (col, val) in self.fixed_parameters.items() if col not in df.columns
+            }
+            new_columns["udp_id"] = self.udp_id
+            new_columns["udp_namespace"] = self.udp_namespace
+            print(new_columns)
+            df = df.assign(**new_columns)
+
+            if len(geojson_params) == 1:
+                #TODO: this is very limited, expand to include more complex cases:
+                # - bbox instead of json
+                if geojson_params[0] not in df.columns:
+                    df.rename_geometry(geojson_params[0],inplace=True)
+            elif len(geojson_params) > 1:
+                for p in geojson_params:
+                    if p not in df.columns:
+                        raise ValueError(f"Multiple geojson parameters, but not all are in the dataframe. Missing column: {p}, available columns: {df.columns}")
+
+            self._persists(df, output_file)
+
+
+
+        def start_job(
+                row: pd.Series,
+                connection: openeo.Connection,
+                **kwargs
+        ) -> openeo.BatchJob:
+
+            def normalize_param_value(name, value):
+                schema = self.udp_parameter_schema(name)
+                if isinstance(value, str) and schema.get("type","") == "array":
+                    return ast.literal_eval( value )
+                elif isinstance(value, str) and schema.get("subtype","") == "geojson":
+                    #this is a side effect of using csv + renaming geometry column
+                    return shapely.geometry.mapping(shapely.wkt.loads(value))
+                else:
+                    return value
+
+            parameters = {k: normalize_param_value(k,row[k]) for k in udp_parameter_names }
+
+
+
+            cube = connection.datacube_from_process(row.udp_id,row.udp_namespace, **parameters)
+
+            title = row.get("title", f"Subjob {row.udp_id} - {str(parameters)}")
```

Comment:
use row index instead of `str(parameters)` in title to avoid extremely large titles?